client_test.go

```
@@ -71,11 +119,246 @@ func TestBaseTags(t *testing.T) {
+// This reproduces that shape: report continuously while Init runs underneath. It is
+// only meaningful under -race, where it fails outright before the fix.
+func TestInitConcurrentWithReporting_IsRaceFree(t *testing.T) {
+	resetClientGlobals(t)
+
+	// A real socket, so the statter is not logging a write error per report and the
+	// test is measuring the config access rather than the failure path.
+	sink, err := net.ListenUDP("udp", &net.UDPAddr{IP: net.ParseIP("127.0.0.1"), Port: 0})
+	if err != nil {
+		t.Fatalf("bind sink: %v", err)
+	}
+	defer sink.Close()
```

golangci-lint (2.13.2)

[error] Error return value of `sink.Close` is not checked

(errcheck)